modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereRequestImpl.java

```diff
@@ -2049,8 +2055,9 @@ private static void load(HttpServletRequest request, Builder b) {
             s = e.nextElement();
             b.headers.put(s, request.getHeader(s));
         }
-
-        e = request.getAttributeNames();
+        synchronized (b.request) {
```

Comment:
No sync is needed on a `get`, no?

Reply:
If it is not possible that a thread invokes the load method while another thread modifies the attributes of the HttpServletRequest, the sync block in the load
get method is not needed. The sync blocks that are needed in my opinion is the ones in method getAttributeNames, setAttribute and removeAttribute.